src/library/scala/Option.scala

```diff
@@ -300,6 +421,14 @@ sealed abstract class Option[+A] extends Product with Serializable {
 
   /** Returns this $option if it is nonempty,
    *  otherwise return the result of evaluating `alternative`.
+   *
+   * This is equivalent to:
+   * {{{
+   * option match {
+   *   case Some(x) => Some(x)
+   *   case None    => Some(alternative)
```

Comment:
this should be `alternative`, not `Some(alternative)`.

There is also some argument to be made of not unboxing and reboxing `x`, but I'm not sure how important that is. The same applies to the comments for `filter` and `filterNot`.